simulariumio/file_converter.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import json
+import logging
+from typing import Any, Dict
+
+from .trajectory_converter import TrajectoryConverter
+from .data_objects import TrajectoryData, UnitData
+
+###############################################################################
+
+log = logging.getLogger(__name__)
+
+###############################################################################
+
+
+class FileConverter(TrajectoryConverter):
+    current_trajectory_info_version: int = 2
+
+    def __init__(self, input_path: str):
+        """
+        This object loads the data in .simularium JSON format
+        at the input path
+
+        Parameters
+        ----------
+        input_path: str
+            path to the .simularium JSON file to load
+        """
+        print("Reading Simularium JSON -------------")
+        with open(input_path) as simularium_file:
+            buffer_data = json.load(simularium_file)
+        if (
+            int(buffer_data["trajectoryInfo"]["version"])
+            < self.current_trajectory_info_version
+        ):
+            buffer_data = self.update_trajectory_info_version(buffer_data)
+        self._data = TrajectoryData.from_buffer_data(buffer_data)
+
+    def update_trajectory_info_version(self, data: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Update the trajectory info block
+        to match the current version
+        """
+        if int(data["trajectoryInfo"]["version"]) == 1:
+            spatial_units = UnitData(
+                "m", data["trajectoryInfo"]["spatialUnitFactorMeters"]
+            )
+            data["trajectoryInfo"]["spatialUnits"] = {
+                "magnitude": spatial_units.magnitude,
+                "name": spatial_units.name,
+            }
+            data["trajectoryInfo"].pop("spatialUnitFactorMeters")
+            time_units = UnitData("s", 1.0)
+            data["trajectoryInfo"]["timeUnits"] = {
+                "magnitude": time_units.magnitude,
+                "name": time_units.name,
+            }
+            data["trajectoryInfo"]["version"] = 2
```

Comment:
for this one I made it explicit because this part of the function is converting v1 -> v2, I'll need a new block for converting v2 -> v3 someday when the current version is 3